### PR TITLE
feat: [PL-28740]: added logs for queries that fetch more than 1000 records

### PR DESCRIPTION
**Pull request by bhavya181**
## Describe your changes
added logs for mongo find query that fetches more than 1000 records at a time, so that we can fix those queries and prevent memory outage

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38362)

### CODE REVIEW

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38362)* status: 1 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java` line 148 at r2](https://reviewable.io/reviews/harness/harness-core/38362#-NDbwVUJ0I8Y-0r2YOup:-NDbwVUJ0I8Y-0r2YOuq:b4pnmke) ([raw file](https://github.com/harness/harness-core/blob/65b86b2298eb2d6a98b88bb4741c3eb0f9ed84e9/959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java#L148)):*
> ```Java
> 
>   @Override
>   public <T> List<T> findDistinct(
> ```

Should we have the check here as well?
___
*[`959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java` line 186 at r2](https://reviewable.io/reviews/harness/harness-core/38362#-NDbwgyE48EZsiaPAdXC:-NDbwgyE48EZsiaPAdXD:b-1safkc) ([raw file](https://github.com/harness/harness-core/blob/65b86b2298eb2d6a98b88bb4741c3eb0f9ed84e9/959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java#L186)):*
> ```Java
> 
>   @Override
>   public <T> List<T> findAllAndRemove(Query query, Class<T> entityClass, String collectionName) {
> ```

We should add over here as well I think
___
*[`959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java` line 198 at r2](https://reviewable.io/reviews/harness/harness-core/38362#-NDbwrgt3ZcF7-qcqryH:-NDbwrgt3ZcF7-qcqryI:b-xm8o85) ([raw file](https://github.com/harness/harness-core/blob/65b86b2298eb2d6a98b88bb4741c3eb0f9ed84e9/959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java#L198)):*
> ```Java
> 
>   @Override
>   public <T> MapReduceResults<T> mapReduce(Query query, String inputCollectionName, String mapFunction,
> ```

We should have a check here as well
___
*[`959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java` line 246 at r2](https://reviewable.io/reviews/harness/harness-core/38362#-NDbxCg71Sj5u2vMAJcl:-NDbxCg71Sj5u2vMAJcm:bjksc57) ([raw file](https://github.com/harness/harness-core/blob/65b86b2298eb2d6a98b88bb4741c3eb0f9ed84e9/959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java#L246)):*
> ```Java
> 
>   @Override
>   protected <O> AggregationResults<O> aggregate(Aggregation aggregation, String collectionName, Class<O> outputType,
> ```

We should have a check here as well.

Basically, we need to check every method in MongoTemplate and see which methods has potential to return list results and override and put a check

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38362)* status: 1 file(s) to review. 1 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java` line 148 at r2](https://reviewable.io/reviews/harness/harness-core/38362#-NDbwVUJ0I8Y-0r2YOup:-NDg6YOL7dI_mCKbuip5:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/65b86b2298eb2d6a98b88bb4741c3eb0f9ed84e9/959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java#L148)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Should we have the check here as well?
</blockquote></details>

Done.
___
*[`959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java` line 186 at r2](https://reviewable.io/reviews/harness/harness-core/38362#-NDbwgyE48EZsiaPAdXC:-NDg6Yqd1Cvg9_7mflz6:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/65b86b2298eb2d6a98b88bb4741c3eb0f9ed84e9/959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java#L186)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

We should add over here as well I think
</blockquote></details>

Done.
___
*[`959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java` line 198 at r2](https://reviewable.io/reviews/harness/harness-core/38362#-NDbwrgt3ZcF7-qcqryH:-NDg6ZN31DaajxZrtFBW:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/65b86b2298eb2d6a98b88bb4741c3eb0f9ed84e9/959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java#L198)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

We should have a check here as well
</blockquote></details>

Done.
___
*[`959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java` line 246 at r2](https://reviewable.io/reviews/harness/harness-core/38362#-NDbxCg71Sj5u2vMAJcl:-NDg6Zv70N4e02ZBeLqD:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/65b86b2298eb2d6a98b88bb4741c3eb0f9ed84e9/959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java#L246)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

We should have a check here as well.

Basically, we need to check every method in MongoTemplate and see which methods has potential to return list results and override and put a check 
</blockquote></details>

Done.

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38362)* status: 1 file(s) to review. 1 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java` line 222 at r4](https://reviewable.io/reviews/harness/harness-core/38362#-NDhAR7wBcyfFi9g_pcG:-NDhAR7wBcyfFi9g_pcH:bnnz6c2) ([raw file](https://github.com/harness/harness-core/blob/622d444df5658a1a7d4759bf2be5ae277426cbe7/959-ng-persistence/src/main/java/io/harness/springdata/HMongoTemplate.java#L222)):*
> ```Java
>       MapReduceResults<T> mapReduceResults =
>           super.mapReduce(query, inputCollectionName, mapFunction, reduceFunction, mapReduceOptions, entityClass);
>       List<Iterator<T>> list = Arrays.asList(mapReduceResults.iterator());
> ```

Can we use mapReduceResults.getMapReduceCounts().getOutputCount()

**Review by ankit-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38362)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38362#-:-NDlbKhQ87WfIvOOIM3-:bnfp4nl)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by rathodmeetsatish — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38362)* status: :shipit: complete! ✓
